packages/atclient/src/monitor.c

```diff
@@ -212,11 +213,21 @@ bool atclient_monitor_is_connected(atclient *monitor_conn) {
 // given a string notification (*original is assumed to JSON parsable), we can deduce the message_type (e.g. data,
 // error, notification) and return the message body which is everything after the prefix (data:, error:, notification:).
 // This function will modify *message_type and *message_body to point to the respective values in *original.
-static int parse_message(char *original, char **message_type, char **message_body) {
+static int parse_message(char *original, size_t original_len, char **message_type, char **message_body) {
   int ret = -1;
   char *temp = NULL;
   char *saveptr;
 
+  size_t read_i;
+  ret = atclient_utils_find_index_past_at_prompt((unsigned char *)original, original_len, &read_i);
+  if (ret != 0) {
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_DEBUG, "Failed to parse the message: %s\n", original);
+    goto exit;
+  }
+  original = original + read_i;
+  original_len = original_len - read_i;
+  original[original_len - 1] = '\0';
+
```

Comment:
Strip prompts from the monitor message, this can happen when the last thing sent or received over the connection was not a notification.

i.e. the first notification after sending monitor or another verb.